### ISSUE TITLE
Embedding webvtt subtitles/automatic captions sometimes causes video corruption

### ISSUE THREAD

**Issue by SunSerega** [yt-dlp/yt-dlp#6918]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that a **supported** site is broken
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

This video: https://www.youtube.com/watch?v=KR-eV7fHNbM
Is downloading corrupted, if all 3 of these are present:
- `--write-auto-sub`
- `--embed-subs`
- `--embed-thumbnail`

At 0:53 it jumps a couple of seconds forward when played in MPV.

I did not try to remove the rest of the options, but I don't think they contribute to the problem.

---

Version:
```
yt-dlp.exe -U
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-o', '[%(upload_date)s] %(title)s.mkv', '-vU', '--ffmpeg-location', 'C:\\Program Files\\ffmpeg\\bin', '--write-auto-sub', '--embed-subs', '--embed-thumbnail', '-f', 'bestvideo+bestaudio', 'https://www.youtube.com/watch?v=KR-eV7fHNbM']
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-02-22-git-d5cc7acff1-full_build-www.gyan.dev (setts), ffprobe 2023-02-22-git-d5cc7acff1-full_build-www.gyan.dev, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[youtube] Extracting URL: https://www.youtube.com/watch?v=KR-eV7fHNbM
[youtube] KR-eV7fHNbM: Downloading webpage
[youtube] KR-eV7fHNbM: Downloading android player API JSON
[debug] [youtube] Extracting signature function js_d87d581f_108
[debug] Loading youtube-sigfuncs.js_d87d581f_108 from cache
[debug] Loading youtube-nsig.d87d581f from cache
[debug] [youtube] Decrypted nsig oDX3iGVOFs0TT9Gpl => BGlBot06RWH7-A
[info] KR-eV7fHNbM: Downloading subtitles: en-zh-TW
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] KR-eV7fHNbM: Downloading 1 format(s): 303+251
[info] Writing video subtitles to: [20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv.en-zh-TW.vtt
[debug] Invoking http downloader on "https://www.youtube.com/api/timedtext?v=KR-eV7fHNbM&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1682528203&sparams=ip%2Cipbits%2Cexpire%2Cv%2Cxoaf&signature=652469E9122BDBB939147867BABE8379C09A0FC9.B366EA2A813EE1200AE44C8B158F4EF9FB8A86C3&key=yt8&lang=zh-TW&tlang=en-zh-TW&fmt=vtt"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: [20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv.en-zh-TW.vtt
[download] 100% of    2.36KiB in 00:00:00 at 11.28KiB/s
[info] Downloading video thumbnail 41 ...
[info] Writing video thumbnail 41 to: [20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv.webp
WARNING: webm doesn't support embedding a thumbnail, mkv will be used
[debug] Invoking dashsegments downloader on "https://rr4---sn-bpb5oxu-3c2r.googlevideo.com/videoplayback?expire=1682524604&ei=XPVIZJ2MJMGLyAXqjYWgBw&ip=2a09%3Abac1%3A7540%3A50%3A%3A49%3Aa0&id=o-AEStJj7nQwHkwH55dn4q3S0mBy9-FMEtk5bQCoU3lNuT&itag=303&source=youtube&requiressl=yes&mh=ZD&mm=31%2C29&mn=sn-bpb5oxu-3c2r%2Csn-3c27sn7e&ms=au%2Crdu&mv=m&mvi=4&pl=64&initcwndbps=995000&spc=qEK7B_tpWw1hsyF1SK-Laal0fYLSkBI&vprv=1&svpuc=1&mime=video%2Fwebm&gir=yes&clen=28968573&dur=234.083&lmt=1681074434441857&mt=1682502797&fvip=3&keepalive=yes&fexp=24007246&c=ANDROID&txp=4535434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgQ9Vzjni5LA5EQV12OdJUIh-T4ftB2P5gAiPRCf7To08CIQCwbq9B-KcNx5mqHTqMs3CdbjsAvlp4Mo2q8_i6rIjjvw%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgYoggtGJy8QRAXfVT25ka6IsEN2HNB662troUxJdj7IUCIQDPwBBUOOhda0LUEBAXekI4qX_QtlsNuOcPF7m7s6lIGQ%3D%3D"
[dashsegments] Total fragments: 3
[download] Destination: [20160331] TheFatRat - The Calling (feat. Laura Brehm).f303.webm
[download] 100% of   27.63MiB in 00:00:40 at 690.02KiB/s
[debug] Invoking dashsegments downloader on "https://rr4---sn-bpb5oxu-3c2r.googlevideo.com/videoplayback?expire=1682524604&ei=XPVIZJ2MJMGLyAXqjYWgBw&ip=2a09%3Abac1%3A7540%3A50%3A%3A49%3Aa0&id=o-AEStJj7nQwHkwH55dn4q3S0mBy9-FMEtk5bQCoU3lNuT&itag=251&source=youtube&requiressl=yes&mh=ZD&mm=31%2C29&mn=sn-bpb5oxu-3c2r%2Csn-3c27sn7e&ms=au%2Crdu&mv=m&mvi=4&pl=64&initcwndbps=995000&spc=qEK7B_tpWw1hsyF1SK-Laal0fYLSkBI&vprv=1&svpuc=1&mime=audio%2Fwebm&gir=yes&clen=3954025&dur=234.101&lmt=1681067543682121&mt=1682502797&fvip=3&keepalive=yes&fexp=24007246&c=ANDROID&txp=4532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRAIgFhsesv2ox1c6dCIxCpebLBvvvPMiKu1RzLY4p0z1GckCIA99do6lp0HYTOAqT8ZSxlzzXYD4-CRLJuNcokeoMVuH&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgYoggtGJy8QRAXfVT25ka6IsEN2HNB662troUxJdj7IUCIQDPwBBUOOhda0LUEBAXekI4qX_QtlsNuOcPF7m7s6lIGQ%3D%3D"
[dashsegments] Total fragments: 1
[download] Destination: [20160331] TheFatRat - The Calling (feat. Laura Brehm).f251.webm
[download] 100% of    3.77MiB in 00:00:05 at 703.56KiB/s
[Merger] Merging formats into "[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv"
[debug] ffmpeg command line: "C:\Program Files\ffmpeg\bin\ffmpeg" -y -loglevel "repeat+info" -i "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).f303.webm" -i "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).f251.webm" -c copy -map "0:v:0" -map "1:a:0" -movflags "+faststart" "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).temp.mkv"
Deleting original file [20160331] TheFatRat - The Calling (feat. Laura Brehm).f303.webm (pass -k to keep)
Deleting original file [20160331] TheFatRat - The Calling (feat. Laura Brehm).f251.webm (pass -k to keep)
[EmbedSubtitle] Embedding subtitles in "[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv"
[debug] ffmpeg command line: "C:\Program Files\ffmpeg\bin\ffmpeg" -y -loglevel "repeat+info" -i "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv" -i "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv.en-zh-TW.vtt" -map 0 -dn -ignore_unknown -c copy -map "-0:s" -map "1:0" "-metadata:s:s:0" "language=eng" "-metadata:s:s:0" "handler_name=English from Chinese (Taiwan)" "-metadata:s:s:0" "title=English from Chinese (Taiwan)" -movflags "+faststart" "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).temp.mkv"
Deleting original file [20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv.en-zh-TW.vtt (pass -k to keep)
[debug] ffprobe command line: "C:\Program Files\ffmpeg\bin\ffprobe" -hide_banner -show_format -show_streams -print_format json "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv"
[EmbedThumbnail] ffmpeg: Adding thumbnail to "[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv"
[debug] ffmpeg command line: "C:\Program Files\ffmpeg\bin\ffmpeg" -y -loglevel "repeat+info" -i "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv" -map 0 -dn -ignore_unknown -c copy -attach "[20160331] TheFatRat - The Calling (feat. Laura Brehm).mkv.webp" "-metadata:s:3" "mimetype=image/webp" "-metadata:s:3" "filename=cover.webp" -movflags "+faststart" "file:[20160331] TheFatRat - The Calling (feat. Laura Brehm).temp.mkv"
```

**Comment by pukkandan** [comment]
Could you try again with `--update-to nightly`?

**Comment by SunSerega** [comment]
Ok, now the version is nightly@2023.04.26.065325
The problem still reproduces.
Do you want me to dump verbose output for this version too?

**Comment by bashonly** [comment]
This seems to be a combination of 2 external issues:
 - a translation error in Youtube's automatic captions
 - (more importantly) an ffmpeg bug with webvtt subtitles and matroska/webm containers


Examining the downloaded .vtt subs file, there is an empty cue payload at the exact timestamp of the video corruption:
```
00:00:49.620 --> 00:00:51.620
That's the beginning

00:00:54.120 --> 00:00:56.220


00:00:56.220 --> 00:00:58.220
Reason and rhyme of hide and seek

```

The original language (zh-TW) .vtt subs look like this:
```
00:00:49.620 --> 00:00:51.620
這就是起點

00:00:54.120 --> 00:00:56.220
捉迷藏

00:00:56.220 --> 00:00:58.220
原因和押韻
```
@gamer191 dug into the translation and discovered that youtube is translating the original `00:54.120` and `00:56.220` cue payloads as a single English cue payload (`00:00:56.220`), which leaves the `00:54.120` cue block empty. So the first half of this issue is an error on Youtube's end is generating invalid(?) webvtt.

The second half of this issue is what happens after ffmpeg embeds the problematic webvtt.

The video corruption seems to only occur if the bad vtt is embedded into a matroska (mkv) or webm container, and then the video is remuxed again afterwards (such as with `--embed-thumbnail`). 

It is easily reproducible with any webm/mkv video and webvtt subs: try editing one of the webvtt cue payloads and removing the caption text so that it is a blank line. Then embed the subs into the mkv/webm, and remux the video again after that (either manually with ffmpeg or with `--embed-thumb`).

We can reproduce it with one of the youtube extractor test cases even:

 - run `yt-dlp -v --ignore-config --write-auto-sub --skip-download test:youtube_5`
 - open the .vtt file in a text editor and edit one or more of the cue payloads (caption text) to be a blank line
 - then run `yt-dlp -v --ignore-config --write-auto-sub test:youtube_5 --no-overwrites --embed-sub --embed-thumb` 

You should see the corruption occur at whatever timestamp you edited the subs file. The filesize will actually decrease slightly too, as if ffmpeg is trimming all streams of that section with the blank subtitles out of the video.

gamer191 tested several versions of ffmpeg dating back to 2021-05-31 and they all had the same issue. mp4 containers are not affected. Converting the subtitles before embedding will prevent the corruption.

**TL;DR:**
 - A workaround for this issue is to add `--convert-sub srt` or `--convert-sub ass` to your command
 - As far as I can tell, the ffmpeg bug has not been reported, which someone should probably do

**Comment by SunSerega** [comment]
Yeah, the workaround works with all the rest of my original options too.
The explanation was also very nice, but I think my understanding is still surface-level.
So please forward this to the FFmpeg bug tracker for me, I think you'd describe it better.

**Comment by pukkandan** [comment]
Closing since there is nothing for us to do. Feel free to continue any discussion.

**Comment by v3DJG6GL** [comment]
I opened a ticket a few weeks ago: https://trac.ffmpeg.org/ticket/11493
Someone recently fixed the bug. One of the next ffmpeg versions should contain the fix :)

**Comment by gamer191** [comment]
> > So please forward this to the FFmpeg bug tracker for me, I think you'd describe it better.
> 
> Will do :)

I apologise, I completely forgot about this. Thanks for reporting it @v3DJG6GL 

I can confirm that this is fixed in the latest ffmpeg master. I'll backport it to https://github.com/yt-dlp/FFmpeg-Builds tomorrow (feel free to remind me if I forget)
